TASK-46: Refresh Access Token after expiration

bit172 opened the pull request:
This PR closes #170.

**Steps to test**
1. Start the application using `nx serve mull-ui` & `nx serve mull-api`.
2. Delete the `mullToken` cookie from your browser if it is present.
3.  Head to the `auth.service.ts` and change the expiration time of `createAccessToken` to `15s` or any other duration to make testing easier.
4. Open the application with the developer tools on the network tab.
5. When making a request after the access token expires, there should be a refresh request.
6. There should be no indication that something went wrong and the request should work as intended.

bit172 committed:
fix(app.tsx): token refresh runs once only

Ref: #170

bit172 committed:
refactor(main.tsx): move apollo client config to module

Ref: #170

bit172 committed:
feat(mull-ui): add automated token refresh upon expiration

Ref: #170

bit172 committed:
chore(mull): add types for apollo-upload-client

Ref: #170

bit172 committed:
doc(apollo-client.ts): add docs to apollo client configs

Ref: #170

RGPosadas reviewed (approved):
Great work, Jon! Everything works seamlessly. LGTM 🚀 
Though quick note (but not a blocker), after running `npm i` with npm version 7.5.3, I ended up with a different package-lock.json. 

**Acceptance Criteria**
- [x] Access Token is refreshed seamlessly when it expires

r-saba commented on apps/mull-ui/src/apollo/apollo-client.ts:
```diff
@@ -0,0 +1,101 @@
+});
+
+const apolloClient = new ApolloClient({
+  link: ApolloLink.from([tokenRefreshLink, authLink, httpLink, uploadLink]),
```
Noticed I can't upload images. It has to do with how these links are setup. `httpLink` is a terminating link so `uploadLink` is never reached. We could remove `httpLink` and use `uploadLink` as our terminating link which sends and receives requests from our server.

bit172 replied:
Good catch, I've decided to use `uploadLink` since it replaces the http link.

bit172 committed:
chore: update package-lock.json

Ref: #170

r-saba reviewed (changes requested):
Refresh works well, noticed an issue with uploading images. Left a comment below

bit172 committed:
fix(apollo-client.ts): remove http link

httplink and upload link are both terminating links. Therefore, if  we're using upload in our application we can simply use the upload link only.
https://github.com/jaydenseric/apollo-upload-client/issues/133

Ref: #170

r-saba commented on apps/mull-ui/src/apollo/apollo-client.ts:
```diff
@@ -0,0 +1,99 @@
+ * This link is responsible for binding the authorization header
+ * on requests.
+ *
+ * It also provides attaches an observable to requests to re-run them
```
small typo
```suggestion
 * It also attaches an observable to requests to re-run them
```

JimLau49 reviewed (approved):
Refresh working good, no other suggestions from me

r-saba reviewed (approved):
LGTM! Thanks!

bit172 committed:
doc(apollo-client.ts): fix typo
Rewrote doc of upload link

Ref: #170
